CRM/Utils/SQL/Select.php

```diff
@@ -330,6 +354,53 @@ public function limit($limit, $offset = 0) {
     return $this;
   }
 
+  /**
+   * Add a union to the list of set operations.
+   *
+   * Ex: CRM_Utils_SQL_Select::fromSet()->union([$subQuery1, $subQuery2])
+   * Ex: CRM_Utils_SQL_Select::fromSet()->union($subQuery1)->union($subQuery2);
+   *
+   * @param string $type "DISTINCT"|"ALL"
+   * @param CRM_Utils_SQL_Select[]|CRM_Utils_SQL_Select $subQueries
+   * @return $this
+   */
+  public function union(string $type, $subQueries) {
+    return $this->setOp("UNION $type", $subQueries);
+  }
+
+  /**
+   * Add a set operation.
+   *
+   * Ex: CRM_Utils_SQL_Select::fromSet()->setOp('INTERSECT', [$subQuery1, $subQuery2])
+   *
+   * @param string $setOperation
+   *   Ex: 'UNION DISTINCT', 'UNION ALL'.
+   *   TODO: 'INTERSECT', 'EXCEPT' when moving to MySQL 8.
+   * @param CRM_Utils_SQL_Select[]|CRM_Utils_SQL_Select $subQueries
+   * @return $this
+   * @see https://dev.mysql.com/doc/refman/8.0/en/set-operations.html
+   */
+  public function setOp(string $setOperation, $subQueries) {
+    // TODO: Support more ops like 'INTERSECT' & 'EXCEPT'
+    $supportedOps = ['UNION DISTINCT', 'UNION ALL'];
```

Comment:
Right, OK, this validation is important because APIv4 allows remote-callers to pass-in the operation. :+1: 

In chat, there was a secondary discussion about whether to allow downstream (*extensions or users*) to call newer ops (`INTERSECT`, `EXCEPT`) even if those ops aren't universally supported. I'm not sure that was resolved - but it's academic right now. When/if someone wants `INTERSECT`, then they'll see where to send a patch. :smile:

Reply:
Yes and I prefer case-sensitive checks for the sake of uniformity. Since there's no *reason* to support lowercase sql keywords in APIv4, we don't, and I'm so glad, it makes parsing everything 200% easier (e.g. `[$expr, $alias] = explode(' AS ', $sql)` is all over the place and is so much cleaner when we just don't support any other variations).